doc/en/user/source/rest/api/stylegenerator.rst

```diff
@@ -0,0 +1,125 @@
+.. _rest_api_stylegenerator:
+
+Style Generator
+===============
+
+The Style Generator endpoint is used for generating default styles. Styles generated using this endpoint may be added to geoserver using the :ref:`styles <rest_api_styles>` endpoint.
+
+``/stylegenerator[.<format>]``
+------------------------------
```

Comment:
Should be able to determine from the `<format>` included in the PUT url.

I actually prefer this approach, creating a REST style generator is confusing as it does not actual modify the geoserver configuration (so does not really belong with the other REST API endpoints).